gateway/coprocess_bundle.go

```diff
@@ -47,22 +47,21 @@ func (b *Bundle) Verify() error {
 	}).Info("----> Verifying bundle: ", b.Spec.CustomMiddlewareBundle)
 
 	var useSignature bool
-	var bundleVerifier goverify.Verifier
+	bundleVerifier := b.Gw.NotificationVerifier
 
 	// Perform signature verification if a public key path is set:
 	if b.Gw.GetConfig().PublicKeyPath != "" {
 		if b.Manifest.Signature == "" {
 			// Error: A public key is set, but the bundle isn't signed.
 			return errors.New("Bundle isn't signed")
 		}
-		if b.Gw.NotificationVerifier == nil {
+		if bundleVerifier == nil {
```

Comment:
What if bundleVerifier is null and the LoadPublicKeyFromFile returns null? useSignature will still be set to true. This may be considered an exceptional flow.

Reply:
Thanks for the comment, @bhuvanaindukuri !
Checking the `goverify` package, `LoadPublicKeyFromFile` only returns nil if there is an error. In that case, we're checking if err != nil, so `useSignature` wouldn't be set to true. 
Please. let me know if I'm missing something.

Reply:
Checked the goverify package and the scenario not seems possible as you mentioned. Thanks for the update.